users/views.py

```diff
@@ -40,6 +42,16 @@ def get_object(self):
         # NOTE: this may be a logged in or anonymous user
         return self.request.user
 
+    def update(self, request, *args, **kwargs):
+        with transaction.atomic():
+            user_name = request.user.name
+            user_update_response = super(CurrentUserRetrieveUpdateViewSet, self).update(
+                request, *args, **kwargs
+            )
```

Comment:
Nitpick: "response" might make the reader think that it's an HTTP response or something like that, and it's best to put the `super` call right at the top so the reader knows right away that this override is adding functionality after the base implementation.

```suggestion
    def update(self, request, *args, **kwargs):
        with transaction.atomic():
            update_result = super(CurrentUserRetrieveUpdateViewSet, self).update(
                request, *args, **kwargs
            )
            user_name = request.user.name
```

Reply:
Suggested changes have been incorporated. There is just one thing that `user_name = request.user.name` is just used to save the old name of the user before super, as super call updates  `request.user.name` value which we use to detect a change in the old and new name. Please suggest if your thoughts on this.